crypto/x509/x_ietfatt.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/err.h>
+#include <openssl/asn1t.h>
+#include <openssl/x509v3.h>
+
+struct IETF_ATTR_SYNTAX_VALUE_st {
+    int type;
+    union {
+        ASN1_OCTET_STRING *octets;
+        ASN1_OBJECT *oid;
+        ASN1_UTF8STRING *string;
+    } u;
+};
+
+struct IETF_ATTR_SYNTAX_st {
+    GENERAL_NAMES *policyAuthority;
+    STACK_OF(IETF_ATTR_SYNTAX_VALUE) *values;
+};
+
+ASN1_CHOICE(IETF_ATTR_SYNTAX_VALUE) = {
```

Comment:
Some ASN.1 structure definitons are commented with a copy-paste of ASN.1 definition from RFC.
I may find some example, if you don't get them ?